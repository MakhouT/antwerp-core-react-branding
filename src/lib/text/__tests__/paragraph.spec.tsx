import * as React from "react";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Paragraph", () => {
    it("Renders bold", () => {
        // Act
        const element = (
            <A.Paragraph type={A.ParagraphStyle.Bold}>test</A.Paragraph>);

        // Act and Assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain("u-text-bold");
    });

    it("Renders italics", () => {
        // Act
        const element = (
            <A.Paragraph type={A.ParagraphStyle.Italic}>test</A.Paragraph>);

        // Act and Assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain("u-text-italic");
    });

    it("Renders alerts when the type is alert", () => {
        // Act
        const element = (
            <A.Paragraph type={A.ParagraphStyle.Alert} level={A.Levels.Danger}>test</A.Paragraph>);

        // Act and Assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain("m-alert--danger");
    });

    it("Don't render alert when the type is not alert", () => {
        // Act
        const element = (
            <A.Paragraph level={A.Levels.Danger}>test</A.Paragraph>);

        // Act and Assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).not.toContain("m-alert--danger");
    });

});
