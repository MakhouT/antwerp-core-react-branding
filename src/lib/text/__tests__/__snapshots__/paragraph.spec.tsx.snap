// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Don't render alert when the type is not alert 1`] = `
<p
  className=""
  style={undefined}
>
  test
</p>
`;

exports[`Paragraph Renders alerts when the type is alert 1`] = `
<p
  className="m-alert m-alert--danger"
  style={undefined}
>
  test
</p>
`;

exports[`Paragraph Renders bold 1`] = `
<div
  className="u-text-bold"
  style={undefined}
>
  test
</div>
`;

exports[`Paragraph Renders italics 1`] = `
<div
  className="u-text-italic"
  style={undefined}
>
  test
</div>
`;
